modules/core/shared/src/main/scala-3.0-/eu/timepit/refined/boolean.scala

```diff
@@ -273,6 +273,15 @@ private[refined] trait BooleanInference0 extends BooleanInference1 {
   implicit def conjunctionCommutativity[A, B]: (A And B) ==> (B And A) =
     Inference.alwaysValid("conjunctionCommutativity")
 
+  implicit def substitutionInConjunction[A, B, C](implicit p1: B ==> C): (A And B) ==> (A And C) =
+    p1.adapt("substitutionInConjunction(%s)")
+
+  implicit def disjunctionTautologyElimination[A, B, C](implicit
+      p1: B ==> C,
```

Comment:
why this ordering? does `B ==> C` need to go first?

Reply:
nope, will swap ;)